include/forcing/ForcingsEngine.hpp

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <vector>
+#include <boost/core/span.hpp>
+
+#include "Bmi_Py_Adapter.hpp"
+#include <utilities/mdarray.hpp>
+
+namespace data_access {
+
+struct ForcingsEngine
+{
+    using size_type                 = std::size_t;
+    using clock_type                = std::chrono::system_clock;
```

Comment:
Eventually we'll want to switch the time keeping to something a bit more thoughtful, ideally UTC-based, but this is fine for now.